test/sequential/test-heapdump-flag.js

```diff
@@ -0,0 +1,28 @@
+// Flags: --heapsnapshot-signal=SIGUSR2
+'use strict';
+const common = require('../common');
+
+if (!common.isMainThread)
+  common.skip('process.chdir is not available in Workers');
+
+const assert = require('assert');
+const fs = require('fs');
+const tmpdir = require('../common/tmpdir');
+
+assert.strictEqual(process.listenerCount('SIGUSR2'), 1);
+tmpdir.refresh();
+process.chdir(tmpdir.path);
+process.kill(process.pid, 'SIGUSR2');
+
+// Asynchronously wait for the snapshot. In development, a single setImmediate()
+// appears to be enough. Use an async loop to be a bit more robust in case
+// platform or machine differences throw off the timing.
+(function validate() {
+  const files = fs.readdirSync(tmpdir.path);
+
+  if (files.length === 0)
+    return setImmediate(validate);
+
+  assert.strictEqual(files.length, 1);
+  assert(/^Heap\..+\.heapsnapshot$/.test(files[0]));
```

Comment:
Consider sending the signal again to prove multiple snapshots are possible, and perhaps read enough of the snapshot to prove it is actually a .heapsnapshot.